### DISCUSSION TITLE
A repl for Odin?

### DISCUSSION THREAD

**Discussion by Inc0n** [odin-lang/Odin#2237] [Ideas/Requests]
Idea: implement a repl for Odin.

References:
https://clang.llvm.org/docs/ClangRepl.html
https://github.com/root-project/cling

**Comment by Kelimion** [comment]
No.

**Reply by Inc0n** [reply]
Hey, at least some one interacted with me, thanks for that. What do you think about just hot code reload? I think that may be a lot easier and more acceptable to get going.

**Reply by Kelimion** [reply]
Hot code reload is already possible using `core:dynlib`.

**Reply by Inc0n** [reply]
NIce.

**Comment by eliasdorneles** [comment]
Okay so, this is terribly hacky and I am embarrassed to share, but I've hacked a Python script that serve as a REPL for simple one-liners: https://github.com/eliasdorneles/dotfiles/blob/master/bin/odinsh.py 😄

What an example session looks like:

```
> ts, utc_offset, _ := time.iso8601_to_time_and_offset("2026-06-03T22:33:00+02:00")
> s, ok := time.time_to_rfc3339(ts, utc_offset)
> s
2026-06-03T22:33:00+02:00
> ok
true
```

It's **VERY** hacky and generally a terrible idea, but may be occasionally useful to quickly try out some function from the core libraries. No promises about maintaining/improving this, but feel free to copy it and make it your own if you so wish! =)

**Comment by gingerBill** [comment]
No.
